packages/compass-connect/src/components/form/form-file-input.jsx

```diff
@@ -2,152 +2,90 @@ import React from 'react';
       [styles['form-item']]: true,
-      [styles['form-item-has-error']]: this.props.error
-    };
-
-    return classnames(className);
-  }
-
-  /**
-   * Gets ReactTooltip id.
-   *
-   * @returns {String} ReactTooltip id.
-   */
-  getErrorId() {
-    return `form-error-tooltip-${this.props.id}`;
-  }
-
-  /**
-   * Opens a tooltip link.
-   */
-  openLink() {
-    shell.openExternal(this.props.link);
-  }
+      [styles['form-item-has-error']]: Boolean(error)
+    });
+  }, [error]);
 
-  /**
-   * Renders a button text.
-   *
-   * @returns {String}
-   */
-  renderButtonText() {
-    if (this.state.values && this.state.values.length > 0) {
-      return this.renderFileNames();
+  const buttonText = React.useMemo(() => {
+    if (Array.isArray(values) && values.length > 0) {
+      return values.map((file) => path.basename(file)).join(', ');
     }
 
-    return this.props.multi ? 'Select files...' : 'Select a file...';
-  }
+    return multi ? 'Select files...' : 'Select a file...';
+  }, [values, multi]);
 
-  /**
-   * Renders file names.
-   *
-   * @returns {String}
-   */
-  renderFileNames() {
-    const baseFiles = this.state.values.map((file) => path.basename(file));
-
-    return baseFiles.join(', ');
-  }
-
-  /**
-   * Renders an info sprinkle.
-   *
-   * @returns {React.Component}
-   */
-  renderInfoSprinkle() {
-    if (this.props.link) {
-      return (
-        <i className={classnames(styles.help)} onClick={this.openLink.bind(this)} />
+  const onFilesChanged = React.useCallback(
+    (evt) => {
+      const files = Array.from(evt.currentTarget.files).map(
+        (file) => file.path
       );
-    }
-  }
-
-  render() {
-    return (
-      <div className={this.getClassName()}>
-        <label>
-          <span>{this.props.label}</span>
-          {this.renderInfoSprinkle()}
-        </label>
-        <Button
-          id={this.props.id}
-          className={styles['form-item-file-button']}
-          onClick={this.onClick.bind(this)}
-          title="Select a file"
-        >
-          <i className="fa fa-upload" aria-hidden />
-          {this.renderButtonText()}
-        </Button>
-      </div>
-    );
-  }
+      changeHandler(files);
+    },
+    [changeHandler]
+  );
+
+  return (
+    <div className={className}>
+      <label htmlFor={id}>
+        <span>{label}</span>
+        {link && (
+          <i
+            className={classnames(styles.help)}
+            onClick={() => shell.openExternal(link)}
+          />
+        )}
+      </label>
+      <input
+        ref={inputRef}
+        id={`${id}_file_input`}
+        name={id}
+        type="file"
+        multiple={multi}
+        onChange={onFilesChanged}
+        style={{ display: 'none' }}
+      />
+      <Button
+        id={id}
+        className={styles['form-item-file-button']}
+        onClick={() => {
+          inputRef.current.click();
+        }}
+        title="Select a file"
+      >
+        <i className="fa fa-upload" aria-hidden />
```

Comment:
How do you feel about using `@leafygreen-ui/icon` here? Maybe the `AddFile` https://www.mongodb.design/component/icon/example/ would also be a bit more descriptive than the upload? Also fine how it is, we can address it later as part of the more general overhaul.

Reply:
Yes, that's a great suggestion, will do!

Reply:
Done in f590b4a